### ISSUE TITLE
Base image from previous stage 0 found, using saved tar at path /kaniko/stages/0 

### ISSUE THREAD

**Issue by rawkode** [GoogleContainerTools/kaniko#294]
Trying to build a multi-stage image and getting:

```shell
$ /kaniko/executor --dockerfile=$CI_PROJECT_DIR/Dockerfile --context=$CI_PROJECT_DIR --no-push --target=test

INFO[0173] Base image from previous stage 0 found, using saved tar at path /kaniko/stages/0 
ERRO[0173] open /kaniko/stages/0: no such file or directory 
```

Dockerfile:

```Dockerfile
FROM ubuntu:16.04 AS base

ENV DEBIAN_FRONTEND noninteractive
ENV LC_ALL C.UTF-8

RUN apt update \
    && apt install --yes software-properties-common curl \
    && curl http://10.23.60.3:90/org/pubkey.gpg | apt-key add - \
    && echo "deb http://10.23.60.3:90/org xenial main" >> /etc/apt/sources.list.d/org.list \
    && add-apt-repository ppa:ondrej/php \
    && apt update \
    && apt install --yes \
    ffmpeg \
    libgearman8 \
    libuv1 \
    php-gearman \
    php-gmagick \
    php-igbinary \
    php-memcached \
    php7.2 \
    php7.2-cassandra \
    php7.2-cli \
    php7.2-common \
    php7.2-curl \
    php7.2-event \
    php7.2-gd \
    php7.2-inotify \
    php7.2-intl \
    php7.2-mbstring \
    php7.2-mysql \
    php7.2-pgsql \
    php7.2-rdkafka \
    php7.2-redis \
    php7.2-soap \
    php7.2-tidy \
    php7.2-xml \
    php7.2-zip \
    && apt clean \
    && rm -rf /var/cache/apt

COPY ./opt/php/php.ini /etc/php/7.2/cli/php.ini

FROM base AS development

ENV PS1 " 🐳 \[\033[1;36m\]\W\[\033[0;35m\] # \[\033[0m\]"

COPY --from=composer:1.6 /usr/bin/composer /usr/bin/composer

RUN apt install --yes \
    make \
    phing \
    phantomjs \
    socat

FROM development AS test

ENV ORG_ENV UnitTest

COPY . /code
WORKDIR /code

RUN make test

FROM base AS production

COPY . /code

```

**Comment by priyawadhwa** [comment]
Hey @rawkode , I merged #295 which should fix this bug. I'm going to close this issue, but feel free to reopen if you encounter any more bugs!

**Comment by pluangashish** [comment]
I am having similar problem but kaniko is failing to store the multistage 

Dockerfile
`FROM ubuntu AS base
RUN echo "base"

FROM base AS stage1
RUN echo "stage1"
CMD ["ls -lrth"]`

Kaniko Image Dockerfile
`FROM gcr.io/kaniko-project/executor:latest as kaniko

FROM busybox:latest as busybox

FROM jenkins/inbound-agent:latest-alpine

# By default, the JNLP3-connect protocol is disabled due to known stability
# and scalability issues. You can enable this protocol using the
# JNLP_PROTOCOL_OPTS environment variable:
#
# JNLP_PROTOCOL_OPTS=-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=false
#
# The JNLP3-connect protocol should be enabled on the Master instance as well.

#ENV JNLP_PROTOCOL_OPTS=-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=false

# Disable the JVM PerfDataFile feature by adding `-XX:-UsePerfData` to the
# `JAVA_OPTS` environment variable. Otherwise, a superfluous
# `/tmp/hsperfdata_root` directory will be included in the final Docker image.

ENV JAVA_OPTS -XX:-UsePerfData

# apk and kaniko must be run as root.
USER root

COPY --from=kaniko /kaniko /kaniko
COPY --from=busybox /bin /busybox

ENV PATH=/busybox:$PATH
ENV SSL_CERT_DIR /kaniko/ssl/certs
ENV PATH=/busybox:$PATH
ENV PATH=/busybox:$PATH
ENV DOCKER_CONFIG /kaniko/.docker/
ENV DOCKER_CREDENTIAL_GCR_CONFIG /kaniko/.config/gcloud/docker_credential_gcr_config.json
# Docker volumes include an entry in /proc/self/mountinfo. This file is used
# when kaniko builds the list of whitelisted directories. Whitelisted
# directories are persisted between stages and are not included in the final
# Docker image.
VOLUME /busybox

# The /kaniko directory is whitelisted by default. Its contents are not de-
# leted between stages, nor is it included in the final Docker image.
WORKDIR /kaniko`


Below is the output of jenkins

`13:12:10  + /kaniko/executor --context /home/jenkins/workspace/Junk/multistagedockerfile --dockerfile /home/jenkins/workspace/Junk/multistagedockerfile/Dockerfile --build-arg 'ENVIRONMENT=development' --verbosity debug --no-push --destination 192389191.dkr.ecr.ap-southeast-3.amazonaws.com/jenkins-kaniko-agent-development:41
13:12:10  DEBU[0000] Copying file /home/jenkins/workspace/Junk/multistagedockerfile/Dockerfile to /kaniko/Dockerfile 
13:12:10  DEBU[0000] file at /kaniko/Dockerfile already exists, resetting file ownership to root 
13:12:10  DEBU[0000] Skip resolving path /kaniko/Dockerfile       
13:12:10  DEBU[0000] Skip resolving path /home/jenkins/workspace/Junk/multistagedockerfile 
13:12:10  DEBU[0000] Skip resolving path /cache                   
13:12:10  DEBU[0000] Skip resolving path                          
13:12:10  DEBU[0000] Skip resolving path                          
13:12:10  DEBU[0000] Skip resolving path                          
13:12:10  DEBU[0000] Skip resolving path                          
13:12:10  INFO[0000] Resolved base name ubuntu to base            
13:12:10  INFO[0000] Resolved base name base to stage1            
13:12:10  DEBU[0000] Built stage name to index map: map[base:0 stage1:1] 
13:12:10  INFO[0000] Retrieving image manifest ubuntu             
13:12:10  INFO[0000] Retrieving image ubuntu from registry index.docker.io 
13:12:12  INFO[0002] Built cross stage deps: map[]                
13:12:12  INFO[0002] Retrieving image manifest ubuntu             
13:12:12  INFO[0002] Returning cached image manifest              
13:12:12  INFO[0002] Executing 0 build triggers                   
13:12:12  INFO[0002] Building stage 'ubuntu' [idx: '0', base-idx: '-1'] 
13:12:12  INFO[0002] Unpacking rootfs as cmd RUN echo "base" requires it. 
13:12:12  DEBU[0002] Ignore list: [{/kaniko false} {/etc/mtab false} {/tmp/apt-key-gpghome true} {/var/run false} {/proc false} {/dev false} {/dev/pts false} {/dev/shm false} {/dev/mqueue false} {/sys false} {/sys/fs/cgroup false} {/sys/fs/cgroup/systemd false} {/sys/fs/cgroup/perf_event false} {/sys/fs/cgroup/cpu,cpuacct false} {/sys/fs/cgroup/blkio false} {/sys/fs/cgroup/hugetlb false} {/sys/fs/cgroup/memory false} {/sys/fs/cgroup/cpuset false} {/sys/fs/cgroup/devices false} {/sys/fs/cgroup/pids false} {/sys/fs/cgroup/net_cls,net_prio false} {/sys/fs/cgroup/freezer false} {/busybox false} {/home/jenkins/.jenkins false} {/home/jenkins/agent false} {/etc/hosts false} {/etc/resolv.conf false} {/etc/hostname false} {/proc/bus false} {/proc/fs false} {/proc/irq false} {/proc/sys false} {/proc/sysrq-trigger false} {/proc/acpi false} {/proc/kcore false} {/proc/keys false} {/proc/latency_stats false} {/proc/timer_list false} {/proc/sched_debug false} {/sys/firmware false}] 
13:12:13  DEBU[0002] Not adding /dev because it is ignored        
13:12:13  DEBU[0002] Not adding /etc/hostname because it is ignored 
13:12:13  DEBU[0002] Not adding /etc/hosts because it is ignored  
13:12:13  DEBU[0002] Not adding /etc/resolv.conf because it is ignored 
13:12:13  DEBU[0002] Not adding /proc because it is ignored       
13:12:13  DEBU[0002] Not adding /sys because it is ignored        
13:12:14  DEBU[0004] Not adding /var/run because it is ignored    
13:12:14  INFO[0004] RUN echo "base"                              
13:12:14  INFO[0004] Initializing snapshotter ...                 
13:12:14  INFO[0004] Taking snapshot of full filesystem...        
13:12:15  DEBU[0004] Current image filesystem: map[]              
13:12:15  ERRO[0004] Couldn't eval /usr/lib/libssl.so.3 with link /usr/lib/libssl.so.3 
13:12:15  DEBU[0004] Adding to layer: [/ /lib /usr/lib /usr /usr/lib/ossl-modules /usr/lib/ossl-modules/legacy.so /usr/lib/modules-load.d /usr/lib/libssl.so.3] 
13:12:15  DEBU[0004] Deleting in layer: map[]                     
13:12:15  INFO[0004] Cmd: /bin/sh                                 
13:12:15  INFO[0004] Args: [-c echo "base"]                       
13:12:15  INFO[0004] Running: [/bin/sh -c echo "base"]            
13:12:15  base
13:12:15  INFO[0004] Taking snapshot of full filesystem...        
13:12:15  DEBU[0004] Current image filesystem: map[/:8635c3e565dc545b3680ea5bcbb8018324b94ae7287940fce56bb3fd6a18408b /lib:3fc1258b4c650de1583119d97232f2851d3e99ab62031a87a5d47f9e9bedad5f /usr:dc7596b23cc64594a9954571b5951e2a9c529849e279d091b6f6b8102891e470 /usr/lib:d1eb2d462eb516c8383942b24dde684b0f2446de8fe23a66edc4533274672658 /usr/lib/libssl.so.3:cf13375561cb47d163f185c510a46d4d9deb0ac63f749d1aed628be8e9a069e1 /usr/lib/modules-load.d:b3005600120f686f49531481ec60ef12fa01d62150eb4dd92024b290cc1084c8 /usr/lib/ossl-modules:d5764288c011e0261028afc7a52bed1c115865262f2a935d02411b505b5ed874 /usr/lib/ossl-modules/legacy.so:dc1c67ce0709c1554ca2201821ad4c66bcbb3cae0a49a54c5ac51ae1d0350a49] 
13:12:15  ERRO[0005] Couldn't eval /usr/lib/libcrypto.so.3 with link /usr/lib/libcrypto.so.3 
13:12:15  DEBU[0005] Adding to layer: [/usr/lib/engines-3 /usr/lib/engines-3/capi.so /usr/lib/engines-3/loader_attic.so /usr/lib/engines-3/padlock.so /usr/lib/engines-3/afalg.so /usr/lib/libcrypto.so.3] 
13:12:15  DEBU[0005] Deleting in layer: map[]                     
13:12:15  DEBU[0005] Mapping stage idx 0 to digest sha256:291969c6d2ffa46be786d9d81c8265feff88db491fd53600649188429512bf62 
13:12:15  DEBU[0005] Mapping digest sha256:291969c6d2ffa46be786d9d81c8265feff88db491fd53600649188429512bf62 to cachekey  
13:12:15  INFO[0005] Storing source image from stage 0 at path /kaniko/stages/0 
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // ecsTaskTemplate
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] wrap
[Pipeline] {
[Pipeline] sh
13:17:12  + echo Admin
13:17:12  Admin
[Pipeline] }
[Pipeline] // wrap
[Pipeline] slackSend
13:17:12  Slack Send Pipeline step running, values are - baseUrl: <empty>, teamDomain: avc, channel: #test, color: #FF0000, botUser: false, tokenCredentialId: slack-token, notifyCommitters: false, iconEmoji: <empty>, username: <empty>, timestamp: <empty>
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // ecsTaskTemplate
[Pipeline] End of Pipeline
ERROR: missing workspace /home/jenkins/workspace/Junk/multistagedockerfile on ecs-kaniko-kaniko-agent-axsna09
Finished: FAILURE`
